content/docs/gke/authentication.md

```diff
@@ -0,0 +1,175 @@
+### Adding RBAC Permissions
+If you find you are missing a permission you need, you can grant the missing roles to your service account using
+Kubernetes resources.
+- **Roles** describe the permissions you want to assign. For example, `verbs: ["create"], resources:["deployments"]`
+- **RoleBindings** define a mapping between the `Role`, and a specific service account
+
+By default, `Roles` and `RoleBindings` apply only to resources in a specific namespace, but there are also
+`ClusterRoles` and `ClusterRoleBindings` that can grant access to resources cluster-wide
+
+More information can be found in the 
+[Kubernetes docs](https://kubernetes.io/docs/reference/access-authn-authz/rbac/#role-and-clusterrole).
+
+# In-Cluster Authentication
+
+When you [set up Kubeflow for GCP](/docs/gke/deploy), it will automatically 
+[provision three service accounts](https://www.kubeflow.org/docs/gke/deploy/deploy-cli/#gcp-service-accounts) with different
+privileges. In particular, the `${KFAPP}-user` service account is meant to grant your user services access to GCP. 
+The credentials to this service account can be accessed within the cluster as a
+[Kubernetes secret](https://kubernetes.io/docs/concepts/configuration/secret/).
+
+The secret will have basic access to a limited set of GCP services by default, but more roles can be granted through the
+[GCP IAM console](https://console.cloud.google.com/iam-admin/).
+
+### Authentication from a Pod
+To access the service account from a Pod, you have to do two things:
+
+1. **Mount the secret as a file.** This will give your pod access to your GCP account, 
+so be careful which pods you grant access to.
+1. **Set the `GOOGLE_APPLICATION_CREDENTIALS` environment variable** to point to the service account.
+GCP libraries will use this environment variable to find the service account and authenticate with GCP.
+
+The following YAML describes a Pod that has access to the `${KFAPP}-user` service account:
+```
+apiVersion: v1
+kind: Pod
+metadata:
+  name: mypod
+spec:
+  containers:
+  - name: mypod
+    image: myimage
+    env:
+    - name: GOOGLE_APPLICATION_CREDENTIALS
+      value: "/var/secrets/user-sa.json"
+    volumeMounts:
+    - name: gcp-secret
+      mountPath: "/var/secrets/user-sa.json"
+      readOnly: true
+  volumes:
+  - name: gcp-secret
+    secret:
+      secretName: myappname-user
+```
+
+### Authentication from Kubeflow Pipelines
+In [Kubeflow Pipelines](https://www.kubeflow.org/docs/pipelines/), each step describes a 
+container that is run independently. If you want to grant access for a single step to use
+ one of your service accounts, you can use 
+[`kfp.gcp.use_gcp_secret()`](https://kubeflow-pipelines.readthedocs.io/en/latest/source/kfp.extensions.html#kfp.gcp.use_gcp_secret)
+Examples for how to use this function can be found in the 
+[Kubeflow examples repo](https://github.com/kubeflow/examples/blob/871895c54402f68685c8e227c954d86a81c0575f/pipelines/mnist-pipelines/mnist_pipeline.py#L97).
```

Comment:
Our pattern in the docs is to include a "Next steps" section at the end of the doc, so that people have somewhere to go if they haven't found what they need. How about this:

```
## Next steps

See the [troubleshooting guide](/docs/gke/troubleshooting-gke/) for help with diagnosing and fixing issues you may encounter with Kubeflow on GCP.
```